modules/apps/document-library/document-library-web/src/main/java/com/liferay/document/library/web/internal/portlet/action/EditPreviewLimitMVCActionCommand.java

```diff
@@ -0,0 +1,138 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2023 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+package com.liferay.document.library.web.internal.portlet.action;
+
+import com.liferay.configuration.admin.constants.ConfigurationAdminPortletKeys;
+import com.liferay.document.library.configuration.DLPreviewLimitConfigurationProvider;
+import com.liferay.document.library.constants.DLPreviewLimitConfigurationConstants;
+import com.liferay.document.library.exception.DLPreviewLimitConfigurationException;
+import com.liferay.portal.configuration.metatype.annotations.ExtendedObjectClassDefinition;
+import com.liferay.portal.configuration.persistence.listener.ConfigurationModelListenerException;
+import com.liferay.portal.kernel.exception.PortalException;
+import com.liferay.portal.kernel.portlet.bridges.mvc.BaseMVCActionCommand;
+import com.liferay.portal.kernel.portlet.bridges.mvc.MVCActionCommand;
+import com.liferay.portal.kernel.servlet.SessionErrors;
+import com.liferay.portal.kernel.util.ParamUtil;
+import com.liferay.portal.kernel.util.Validator;
+
+import javax.portlet.ActionRequest;
+import javax.portlet.ActionResponse;
+
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+
+/**
+ * @author Marco Galluzzi
+ */
+@Component(
+	property = {
+		"javax.portlet.name=" + ConfigurationAdminPortletKeys.INSTANCE_SETTINGS,
+		"javax.portlet.name=" + ConfigurationAdminPortletKeys.SITE_SETTINGS,
+		"javax.portlet.name=" + ConfigurationAdminPortletKeys.SYSTEM_SETTINGS,
+		"mvc.command.name=/instance_settings/edit_preview_limit"
+	},
+	service = MVCActionCommand.class
+)
+public class EditPreviewLimitMVCActionCommand extends BaseMVCActionCommand {
+
+	@Override
+	protected void doProcessAction(
+			ActionRequest actionRequest, ActionResponse actionResponse)
+		throws Exception {
+
+		ExtendedObjectClassDefinition.Scope scope = _getScope(actionRequest);
+
+		try {
+			_dlPreviewLimitConfigurationProvider.updatePreviewLimit(
+				_getMaxFileSize(actionRequest),
+				_getMaxNumberOfPages(actionRequest), scope,
+				_getScopePK(actionRequest, scope));
```

Comment:
(optional) Shouldn't this provider method be named `updateDLPreviewLimit`?

Reply:
done